biogeochem/FatesSoilBGCFluxMod.F90

```diff
@@ -0,0 +1,1114 @@
+    ! The purpose of this routine is to recieve the nutrient uptake flux
+    ! boundary conditions, and parse those fluxes out to the cohorts.
+    !
+    ! This routine should be called before FATES dynamics, particularly before
+    ! any of the PARTEH code is called.  It is assumed that these utpake fluxes
+    ! are being incremented each short-BGC timestep over the course of the day, and
+    ! thus should be an integrated quantity, total nutrient uptake, over 1 day.
+    ! At the end of this routine, after we have parsed the uptake to the cohorts,
+    ! we can then zero out the input boundary conditions again, so they can be
+    ! integrated.
+    ! -----------------------------------------------------------------------------------
+
+
+    ! !ARGUMENTS    
+    type(ed_site_type), intent(inout) :: sites(:)
+    type(bc_in_type), intent(in)     :: bc_in(:)
+
+    ! Locals
+    integer                       :: nsites        ! number of sites
+    integer                       :: s             ! site loop index
+    integer                       :: j             ! soil layer
+    integer                       :: icomp         ! competitor index
+    integer                       :: id            ! decomp layer index
+    integer                       :: pft           ! pft index
+    type(ed_patch_type), pointer  :: cpatch        ! current patch pointer
+    type(ed_cohort_type), pointer :: ccohort       ! current cohort pointer
+    real(r8) :: fnrt_c                             ! fine-root carbon [kg]
+    integer  :: nlev_eff_soil                      ! we only operate over
+                                                   ! the effective soil
+                                                   ! layer because we
+                                                   ! assume no frozen uptake
+
+    real(r8), allocatable :: rootfrac_pft_decomp(:,:) ! fraction of root for each
+                                                      ! pft in each decomp layer
+    real(r8), allocatable :: fnrt_c_pft(:)            ! total mass of root for each PFT [kgC]
+
+
+    nsites = size(sites,dim=1)
+
+    
+    ! Zero the uptake rates
+    do s = 1, nsites
+       cpatch => sites(s)%oldest_patch
+       do while (associated(cpatch))
+          ccohort => cpatch%tallest
+          do while (associated(ccohort))
+             ccohort%daily_n_uptake = 0._r8
+             ccohort%daily_p_uptake = 0._r8
+             ccohort => ccohort%shorter
+          end do
+          cpatch => cpatch%younger
+       end do
+       
+    end do
+    
+    ! We can exit if this is a c-only simulation
+    if(hlm_parteh_mode.ne.prt_cnp_flex_allom_hyp) then
+       ! These can now be zero'd
+       do s = 1, nsites
+          bc_in(s)%plant_n_uptake_flux(:,:) = 0._r8
+          bc_in(s)%plant_p_uptake_flux(:,:) = 0._r8
+       end do
+       return
+    end if
+
+    
+    do s = 1, nsites
+
+       ! If the plant's PFT has a non-zero positive prescribed rate,
```

Comment:
I'm struggling to map this comment into the IF statement below...

Reply:
updated comment:
```   
       ! If the plant is in "prescribed uptake mode"
       ! then we are not coupling with the soil bgc model.
       ! In this case, the bc_in structure is meaningless.
       ! Instead, we give the plants a parameterized fraction
       ! of their demand.  Routine GetPlantDemand() returns
       ! the plant demand.
```